### DISCUSSION TITLE
Setting visibility of generated types to internal

### DISCUSSION THREAD

**Discussion by vytskalt** [square/wire#2771] [General]
Hi, is it possible to make the generated .kt types not visible to other modules? I have a module in which I am using gRPC but it is an implementation detail and the generated types are not supposed to be used by other modules, so it would be great to somehow hide them

**Comment by JakeWharton** [comment]
I would recommend generating the protos in one module and then using that as an `implementation` dependency in another. Then consuming modules of that other module which does gRPC will not see any of the proto types.

**Reply by vytskalt** [reply]
Thanks for the idea
